package/kedro_viz/server.py

```diff
@@ -70,35 +75,35 @@ def pretty_name(name):
     all_tags = set()
 
     for node in sorted(pipeline.nodes, key=lambda n: n.name):
-        task_id = "task/" + node.name.replace(" ", "")
+        task_id = _hash(str(node))
         nodes.append(
             {
                 "type": "task",
                 "id": task_id,
                 "name": getattr(node, "short_name", node.name),
-                "full_name": str(node),
+                "full_name": getattr(node, "_func_name", str(node)),
```

Comment:
If 2 nodes are using the same underlying function, do we expect them to have the same `full_name`?

Reply:
They will if names have not been manually specified, that is fine because they will still have different inputs and outputs in the diagram